cpg_workflows/stages/dragen_ica/run_align_genotype_with_dragen.py

```diff
@@ -0,0 +1,125 @@
+    cram_index_id: str,
+    dragen_ht_id: str,
+    cram_reference_id: str,
+    dragen_pipeline_id: str,
+    ica_output_folder_id: str,
+    user_tags: list[str],
+    technical_tags: list[str],
+    reference_tags: list[str],
+    user_reference: str,
+    project_id: dict[str, str],
+    api_instance: project_analysis_api.ProjectAnalysisApi,
+) -> str:
+    header_params: dict[Any, Any] = {}
+    body = CreateNextflowAnalysis(
+        userReference=user_reference,
+        pipelineId=dragen_pipeline_id,
+        tags=AnalysisTag(
+            technicalTags=technical_tags,
+            userTags=user_tags,
+            referenceTags=reference_tags,
+        ),
+        outputParentFolderId=ica_output_folder_id,
+        analysisInput=NextflowAnalysisInput(
+            inputs=[
+                AnalysisDataInput(parameterCode='crams', dataIds=[cram_id]),
+                AnalysisDataInput(parameterCode='crais', dataIds=[cram_index_id]),
+                AnalysisDataInput(parameterCode='ref_tar', dataIds=[dragen_ht_id]),
+                AnalysisDataInput(parameterCode='cram_reference', dataIds=[cram_reference_id]),
+            ],
+            parameters=[
+                AnalysisParameterInput(
+                    code='enable_map_align',
+                    value='True',
+                ),
+                AnalysisParameterInput(code='enable_map_align_output', value='True'),
+                AnalysisParameterInput(code='output_format', value='CRAM'),
+                AnalysisParameterInput(code='enable_duplicate_marking', value='True'),
+                AnalysisParameterInput(code='enable_variant_caller', value='True'),
+                AnalysisParameterInput(code='vc_emit_reference_confidence', value='GVCF'),
+                AnalysisParameterInput(code='vc_enable_vcf_output', value='False'),
+                AnalysisParameterInput(code='enable_cnv', value='True'),
+                AnalysisParameterInput(code='enable_sv', value='True'),
+                AnalysisParameterInput(
+                    code='additional_args',
+                    value='--vc-hard-filter "DRAGENHardQUAL:all:QUAL<5.0;LowDepth:all:DP<=1" --vc-frd-max-effective-depth 40 --vc-enable-joint-detection true',
+                ),
+            ],
+        ),
+    )
+    try:
+        api_response = api_instance.create_nextflow_analysis(
+            path_params=project_id,
+            header_params=header_params,
+            body=body,
+        )
+        return api_response.body['id']
+    except icasdk.ApiException as e:
+        raise icasdk.ApiException(f'Exception when calling ProjectAnalysisApi->create_nextflow_analysis: {e}') from e
+
+
+def run(
+    ica_fids_path: str,
+    analysis_output_fid_path: str,
+    dragen_ht_id: str,
+    cram_reference_id: str,
+    dragen_pipeline_id: str,
+    user_tags: list[str],
+    technical_tags: list[str],
+    reference_tags: list[str],
+    user_reference: str,
+    api_root: str,
+) -> dict[str, str]:
```

Comment:
If I'm being a picky bugger, seeing as so many of these methods are all called `run`, a docstring explanation of what this `run` runs would be cool. In the job call it's clearer as `run_align_genotype_with_dragen.run` but in this review I keep bouncing up to the file name to know what the method does

Reply:
Yeah can do this